packages/entities/entities-plugins/src/components/PluginEntityForm.vue

```diff
@@ -0,0 +1,717 @@
+        :is="sharedFormName"
+        v-if="sharedFormName && (formModel.id && isEditing || !isEditing)"
+        :form-model="formModel"
+        :form-options="formOptions"
+        :form-schema="formSchema"
+        :is-editing="isEditing"
+        :on-model-updated="onModelUpdated"
+      />
+
+      <VueFormGenerator
+        v-if="!sharedFormName && (formModel.id && isEditing || !isEditing)"
+        :model="formModel"
+        :options="formOptions"
+        :schema="formSchema"
+        @model-updated="onModelUpdated"
+      />
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { computed, ref, reactive, provide, watch, type PropType, onBeforeMount, defineComponent } from 'vue'
+import type { AxiosResponse, AxiosRequestConfig } from 'axios'
+import {
+  EntityTypeIdField,
+  type PluginScope,
+  type KonnectPluginFormConfig,
+  type KongManagerPluginFormConfig,
+} from '../types'
+import { useAxios } from '@kong-ui-public/entities-shared'
+import {
+  customFields,
+  getSharedFormName,
+  FORMS_API_KEY,
+  sharedForms,
+} from '@kong-ui-public/forms'
+import '@kong-ui-public/forms/dist/style.css'
+import composables from '../composables'
+import endpoints from '../plugins-endpoints'
+
+// Must explicitly specify these as components since they are rendered dynamically
+export default defineComponent({
+  components: { ...sharedForms },
+})
+</script>
+
+<script setup lang="ts">
+const emit = defineEmits<{
+  (e: 'loading', isLoading: boolean): void,
+  (e: 'model-updated',
+    payload: {
+      model: Record<string, any>,
+      originalModel: Record<string, any>,
+      data: Record<string, any>
+    }
+  ): void,
+}>()
+
+const props = defineProps({
+  /** The base konnect or kongManger config. Pass additional config props in the shared entity component as needed. */
+  config: {
+    type: Object as PropType<KonnectPluginFormConfig | KongManagerPluginFormConfig>,
+    required: true,
+    validator: (config: KonnectPluginFormConfig | KongManagerPluginFormConfig): boolean => {
+      if (!config || !['konnect', 'kongManager'].includes(config?.app)) return false
+      if (config.app === 'konnect' && !config.controlPlaneId) return false
+      if (config.app === 'kongManager' && typeof config.workspace !== 'string') return false
+      if (!config.cancelRoute || !config.backRoute) return false
+      return true
+    },
+  },
+  entityId: {
+    type: String,
+    default: '',
+  },
+  entityType: {
+    type: String as PropType<PluginScope>,
+    default: null,
+  },
+  /**
+   * Plugin data if being edited
+   */
+  record: {
+    type: Object as PropType<Record<string, any>>,
+    default: () => ({}),
+  },
+  schema: {
+    type: Object as PropType<Record<string, any>>,
+    default: () => ({}),
+  },
+  isEditing: {
+    type: Boolean,
+    default: false,
+  },
+  isCredential: {
+    type: Boolean,
+    default: false,
+  },
+})
+
+const { axiosInstance } = useAxios({
+  headers: props.config?.requestHeaders,
```

Comment:
`props.config` is required, so you shouldn't need to optionally chain. Same comment for whole PR